packages/core/src/docs/data-model/text-x/build-utils/text-x-utils.ts

```diff
@@ -351,18 +244,47 @@ export const replaceSelectionTextX = (params: IReplaceSelectionTextXParams) => {
     const body = doc.getSelfOrHeaderFooterModel(segmentId)?.getBody();
     if (!body) return false;
 
+    const oldBody = selection.collapsed ? null : getBodySlice(body, selection.startOffset, selection.endOffset);
+    const diffs = textDiff(oldBody ? oldBody.dataStream : '', insertBody.dataStream);
```

Comment:
This util function has not tests covered.

Reply:
I mean the whole replace function.

Reply:
added